tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml_iteration.bal

```diff
@@ -142,10 +140,103 @@ function xmlSequenceIter() returns string {
 
 function xmlCharItemIter() returns string {
     string result = "";
-
-    foreach xml|string elem in bitOfText {
+    int i = 0;
+    foreach xml elem in bitOfText {
         string str = io:sprintf("%s\n", elem);
         result += str;
+        i += 1;
     }
     return result;
 }
+
+function xmlTypeParamElementIter() {
+    xml<'xml:Element> el1 = xml `<foo>foo</foo><bar/>`;
```

Comment:
These tests are bound to break once we fix https://github.com/ballerina-platform/ballerina-lang/issues/26340, right?

Instead shall we rewrite these tests similar to 

```ballerina
public function main() {
    xml el = xml `<foo>foo</foo>`;
    xml<'xml:Element> seq = <xml<'xml:Element>> el.concat(xml `<bar/>`);


    foreach 'xml:Element x in seq {
        
    }
}
```